californium-core/src/test/java/org/eclipse/californium/core/coap/RequestTest.java

```diff
@@ -67,8 +68,21 @@ public void testSetURISetsDestination() throws UnknownHostException {
 
 	@Test
 	public void testSetURISetsUriHostOption() throws UnknownHostException {
-		Request req = Request.newGet().setURI("coap://iot.eclipse.org");
-		assertThat(req.getOptions().getUriHost(), is("iot.eclipse.org"));
```

Comment:
I would remove the special case for 'localhost' and test with 'localhost' in place of 'iot.eclipse.org'